doc/build-openbsd.md

```diff
@@ -1,123 +1,138 @@
+## Preparation
+
+### 1. Install Required Dependencies
+Run the following as root to install the base dependencies for building.
 
 ```bash
-pkg_add git gmake libevent libtool boost
-pkg_add qt5 # (optional for enabling the GUI)
-pkg_add autoconf # (select highest version, e.g. 2.69)
+pkg_add bash git gmake libevent libtool boost
+pkg_add autoconf # (select highest version, e.g. 2.71)
 pkg_add automake # (select highest version, e.g. 1.16)
-pkg_add python # (select highest version, e.g. 3.8)
-pkg_add bash
+pkg_add python # (select highest version, e.g. 3.9)
+```
+
+See [dependencies.md](dependencies.md) for a complete overview.
 
+### 2. Clone Bitcoin Repo
+Clone the Bitcoin Core repository to a directory. All build scripts and commands will run from this directory.
+``` bash
 git clone https://github.com/bitcoin/bitcoin.git
 ```
 
-See [dependencies.md](dependencies.md) for a complete overview.
+### 3. Install Optional Dependencies
+
+#### Wallet Dependencies
+
+It is not necessary to build wallet functionality to run either `bitcoind` or `bitcoin-qt`.
+
+###### Descriptor Wallet Support
 
-**Important**: From OpenBSD 6.2 onwards a C++11-supporting clang compiler is
-part of the base image, and while building it is necessary to make sure that
-this compiler is used and not ancient g++ 4.2.1. This is done by appending
-`CC=cc CXX=c++` to configuration commands. Mixing different compilers within
-the same executable will result in errors.
+`sqlite3` is required to support [descriptor wallets](descriptors.md).
 
-### Building BerkeleyDB
+``` bash
+pkg_add install sqlite3
+```
 
-BerkeleyDB is only necessary for the wallet functionality. To skip this, pass
-`--disable-wallet` to `./configure` and skip to the next section.
+###### Legacy Wallet Support
+BerkeleyDB is only required to support legacy wallets.
 
 It is recommended to use Berkeley DB 4.8. You cannot use the BerkeleyDB library
-from ports, for the same reason as boost above (g++/libstd++ incompatibility).
-If you have to build it yourself, you can use [the installation script included
-in contrib/](/contrib/install_db4.sh) like so:
+from ports. However you can build it yourself, [using the installation script included in contrib/](/contrib/install_db4.sh), like so, from the root of the repository.
 
 ```bash
-./contrib/install_db4.sh `pwd` CC=cc CXX=c++
+./contrib/install_db4.sh `pwd`
 ```
 
-from the root of the repository. Then set `BDB_PREFIX` for the next section:
+Then set `BDB_PREFIX`:
 
 ```bash
 export BDB_PREFIX="$PWD/db4"
 ```
 
-### Building Bitcoin Core
+#### GUI Dependencies
+###### Qt5
+
+Bitcoin Core includes a GUI built with the cross-platform Qt Framework. To compile the GUI, Qt 5 is required.
+
+```bash
+pkg_add qt5
+```
+
+## Building Bitcoin Core
 
 **Important**: Use `gmake` (the non-GNU `make` will exit with an error).
 
 Preparation:
 ```bash
 
 # Replace this with the autoconf version that you installed. Include only
```

Comment:
2. “Replace this with autoconf version you installed.” Though it need not specifically be part of the documentation, I think we can give a redirect to a webpage explaining how to know the currently installed version of a package in OpenBSD.

Reply:
I think that's something that someone can either look up, or read the OpenBSD documentation for; assuming they've installed the package previously, otherwise if they've just installed the package while running through these docs, they already know the version.

Reply:
I think that's a fair point. So I think we can forgo my 2nd suggestion.

@fanquake 
I want to understand your take on my first suggestion about removing example version numbers.

Reply:
> removing example version numbers.

@shaavan I've pushed a change that should resolve that.